src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx

```diff
@@ -196,16 +213,27 @@ export class FieldsBreakdownScene extends SceneObjectBase<FieldsBreakdownSceneSt
         stateUpdate.body = this.buildEmptyLayout();
       }
     } else {
-      stateUpdate.body =
-        newState.value === ALL_VARIABLE_VALUE
-          ? this.buildFieldsLayout(newState.options.map((opt) => ({ label: opt.label, value: String(opt.value) })))
-          : this.buildValuesLayout(newState);
+      // Otherwise update the body, but don't re-instantiate if it's already the right class
+      if (newState.value === ALL_VARIABLE_VALUE && this.state.body instanceof FieldValuesBreakdownScene) {
+        stateUpdate.body = new FieldsAggregatedBreakdownScene({});
+      } else if (newState.value !== ALL_VARIABLE_VALUE && this.state.body instanceof FieldsAggregatedBreakdownScene) {
+        stateUpdate.body = new FieldValuesBreakdownScene({});
+      } else if (
+        this.state.body === undefined ||
+        this.state.body instanceof SceneFlexLayout ||
+        this.state.body instanceof SceneReactObject
```

Comment:
```
        this.state.body instanceof SceneFlexLayout ||
        this.state.body instanceof SceneReactObject
```

I'm not sure I understand what this means.

Reply:
I added a bit of explanation in a comment here